src/main/scala/org/broadinstitute/dsde/agora/server/webservice/handlers/PermissionHandler.scala

```diff
@@ -1,74 +1,77 @@
 package org.broadinstitute.dsde.agora.server.webservice.handlers
 
 import akka.actor.Actor
+import akka.pattern._
 import org.broadinstitute.dsde.agora.server.business.PermissionBusiness
-import org.broadinstitute.dsde.agora.server.dataaccess.permissions.{AgoraPermissions, AccessControl}
+import org.broadinstitute.dsde.agora.server.dataaccess.permissions.{AccessControl, AgoraPermissions, PermissionsDataSource}
 import org.broadinstitute.dsde.agora.server.dataaccess.permissions.AgoraPermissions._
 import org.broadinstitute.dsde.agora.server.model.AgoraEntity
 import org.broadinstitute.dsde.agora.server.webservice.PerRequest.RequestComplete
 import org.broadinstitute.dsde.agora.server.webservice.util.ServiceMessages._
 import spray.routing.RequestContext
 
+import scala.concurrent.ExecutionContext
 
-class PermissionHandler extends Actor {
+
+class PermissionHandler(dataSource: PermissionsDataSource, implicit val ec: ExecutionContext) extends Actor {
 
   // JSON Serialization Support
   import spray.httpx.SprayJsonSupport._
   import org.broadinstitute.dsde.agora.server.model.AgoraApiJsonSupport._
 
   implicit val system = context.system
-  val permissionBusiness = new PermissionBusiness()
+  val permissionBusiness = new PermissionBusiness(dataSource)
 
   def receive = {
     case ListNamespacePermissions(_context: RequestContext, entity: AgoraEntity, requester: String) =>
-      val permissions = permissionBusiness.listNamespacePermissions(entity, requester)
-      context.parent ! RequestComplete(permissions)
-      context.stop(self)
+      (permissionBusiness.listNamespacePermissions(entity, requester) map { permissions =>
+        RequestComplete(permissions)
+      }) pipeTo context.parent
```

Comment:
i'm not the biggest fan of this pattern of call-to-permissionBusiness-map-result, but it works